graphql/graphql.go

```diff
@@ -820,15 +844,7 @@ func (b *Block) LogsBloom(ctx context.Context) (hexutil.Bytes, error) {
 }
 
 func (b *Block) TotalDifficulty(ctx context.Context) (hexutil.Big, error) {
-	h := b.hash
-	if h == (common.Hash{}) {
-		header, err := b.resolveHeader(ctx)
-		if err != nil {
-			return hexutil.Big{}, err
-		}
-		h = header.Hash()
-	}
-	td := b.r.backend.GetTd(ctx, h)
+	td := b.r.backend.GetTd(ctx, b.hash)
```

Comment:
I don't get this... So on the method `func (b *Block) Hash`, you obtain the lock to acccess `b.hash`. But here, you just access the `b.hash` directly? Shouldn't you use `b.Hash()` instead?

Reply:
Yes will change to `b.Hash()`.